### DISCUSSION TITLE
Involve related spark kyuubi-extension modules into kyuubi-spark-sql-engine.jar

### DISCUSSION THREAD

**Discussion by turboFei** [apache/kyuubi#1751] [Ideas]
For example:
if the profile for building kyuubi is `-Pspark-3.1`.

How about involving modules `kyuubi-extensions-spark-common` and `kyuubi-extensions-spark-3.1` into kyuubi-spark-sql-engine.jar directly?

It should be more convenient.

Any concern for this?

**Comment by yaooqinn** [comment]
If you switch SPARK_HOME in runtime, you may fail

**Comment by pan3793** [comment]
Then the `kyuubi-spark-sql-engine.jar` becomes a version specific, it can not run across spark 3.0/3.1/3.2

**Comment by turboFei** [comment]
Thanks, got it.

I can do that for my use case.

For my use case:

1. we maintain single spark version for kyuubi spark engine, so the spark version is bound to kyuubi server.

**Reply by yaooqinn** [reply]
use spark conf to point to the extension

**Reply by turboFei** [reply]
thanks, will try to use `spark.jars`.

**Reply by pan3793** [reply]
I'm wondering if we can support multi jars when submit engine, it's value for case like mysql jdbc driver, due the license issue, we can not bundle it into engine jar.
